MJML error with v20210212

**Describe the bug**
When attempting to send emails using the email plugin I get the following in my error log. As with #976 I was not defining the mjml path inside my environmental vars. The incident _does_ create successfully, but it does not send out emails. This build is using v20210212 as base. How would you recommend resolving? I have not worked with mjml before. 

```
2021-02-23 13:55:02.637 EST- /usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml

2021-02-23 13:55:02.637 EST at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)

2021-02-23 13:55:02.637 EST at Function.Module._load (internal/modules/cjs/loader.js:862:27)

2021-02-23 13:55:02.637 EST at Module.require (internal/modules/cjs/loader.js:1042:19)

2021-02-23 13:55:02.637 EST at require (internal/modules/cjs/helpers.js:77:18)

2021-02-23 13:55:02.637 EST at Object.<anonymous> (/usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml:3:1)

2021-02-23 13:55:02.638 ESTTraceback (most recent call last): File "/usr/local/lib/python3.8/site-packages/dispatch/decorators.py", line 37, in wrapper result = func(*args, **kwargs) File "/usr/local/lib/python3.8/site-packages/dispatch/incident/flows.py", line 687, in incident_create_flow send_incident_created_notifications(incident, db_session) File "/usr/local/lib/python3.8/site-packages/dispatch/incident/messaging.py", line 266, in send_incident_created_notifications notification_service.filter_and_send( File "/usr/local/lib/python3.8/site-packages/dispatch/notification/service.py", line 116, in filter_and_send send( File "/usr/local/lib/python3.8/site-packages/dispatch/notification/service.py", line 92, in send plugin.instance.send( File "/usr/local/lib/python3.8/site-packages/dispatch/decorators.py", line 58, in wrapper result = func(*args, **kwargs) File "/usr/local/lib/python3.8/site-packages/dispatch/decorators.py", line 74, in wrapper return func(*args, **kwargs) File "/usr/local/lib/python3.8/site-packages/dispatch/plugins/dispatch_google/gmail/plugin.py", line 94, in send message_body = create_message_body(notification_template, notification_type, **kwargs) File "/usr/local/lib/python3.8/site-packages/dispatch/messaging/email/utils.py", line 87, in create_message_body return render_html(template.render(**kwargs)) File "/usr/local/lib/python3.8/site-packages/dispatch/messaging/email/utils.py", line 103, in render_html raise Exception("MJML template processing failed.") Exception: MJML template processing failed.
```

Thank you again for your patience. Here is a copy of the log with the adjusted build. 

```
web_1        | INFO:     172.18.0.1:59982 - "GET /api/v1/tags/ HTTP/1.1" 200 OK
web_1        | INFO:     172.18.0.1:59974 - "GET /api/v1/individuals/ HTTP/1.1" 200 OK
web_1        | INFO:     172.18.0.1:59986 - "GET /api/v1/incidents/ HTTP/1.1" 200 OK
web_1        | ERROR:dispatch.messaging.email.utils:internal/modules/cjs/loader.js:983
web_1        |   throw err;
web_1        |   ^
web_1        | 
web_1        | Error: Cannot find module '../lib/index'
web_1        | Require stack:
web_1        | - /usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml
web_1        |     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
web_1        |     at Function.Module._load (internal/modules/cjs/loader.js:862:27)
web_1        |     at Module.require (internal/modules/cjs/loader.js:1042:19)
web_1        |     at require (internal/modules/cjs/helpers.js:77:18)
web_1        |     at Object.<anonymous> (/usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml:3:1)
web_1        |     at Module._compile (internal/modules/cjs/loader.js:1156:30)
web_1        |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
web_1        |     at Module.load (internal/modules/cjs/loader.js:1000:32)
web_1        |     at Function.Module._load (internal/modules/cjs/loader.js:899:14)
web_1        |     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12) {
web_1        |   code: 'MODULE_NOT_FOUND',
web_1        |   requireStack: [
web_1        |     '/usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml'
web_1        |   ]
web_1        | }
web_1        | 
web_1        | ERROR:dispatch.decorators:MJML template processing failed.
web_1        | Traceback (most recent call last):
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/decorators.py", line 37, in wrapper
web_1        |     result = func(*args, **kwargs)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/report/flows.py", line 71, in create_tactical_report
web_1        |     send_tactical_report_to_tactical_group(incident_id, tactical_report, db_session)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/report/messaging.py", line 82, in send_tactical_report_to_tactical_group
web_1        |     plugin.instance.send(
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/decorators.py", line 58, in wrapper
web_1        |     result = func(*args, **kwargs)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/decorators.py", line 74, in wrapper
web_1        |     return func(*args, **kwargs)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/plugins/dispatch_google/gmail/plugin.py", line 94, in send
web_1        |     message_body = create_message_body(notification_template, notification_type, **kwargs)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/messaging/email/utils.py", line 88, in create_message_body
web_1        |     return render_html(template.render(**kwargs))
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/messaging/email/utils.py", line 104, in render_html
web_1        |     raise Exception("MJML template processing failed.")
web_1        | Exception: MJML template processing failed.
web_1        | INFO:     172.18.0.1:59968 - "GET /api/v1/incidents/?q=&page=1&itemsPerPage=10&sortBy[]=reported_at&descending[]=true HTTP/1.1" 200 OK
```

Looks like Dispatch can't find the `mjml` node module binary in `/usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml`. We will look into it and circle back.

```
web_1        | Error: Cannot find module '../lib/index'
web_1        | Require stack:
web_1        | - /usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml
web_1        |     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
web_1        |     at Function.Module._load (internal/modules/cjs/loader.js:862:27)
web_1        |     at Module.require (internal/modules/cjs/loader.js:1042:19)
web_1        |     at require (internal/modules/cjs/helpers.js:77:18)
web_1        |     at Object.<anonymous> (/usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml:3:1)
web_1        |     at Module._compile (internal/modules/cjs/loader.js:1156:30)
web_1        |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
web_1        |     at Module.load (internal/modules/cjs/loader.js:1000:32)
web_1        |     at Function.Module._load (internal/modules/cjs/loader.js:899:14)
web_1        |     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12) {
web_1        |   code: 'MODULE_NOT_FOUND',
web_1        |   requireStack: [
web_1        |     '/usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml'
web_1        |   ]
web_1        | }
```

I tried the following as an attempt to give full path in case that was the issue.
`MJML_PATH=/usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin` 

and got the following error

```
web_1        | INFO:     172.20.0.1:64258 - "GET /api/v1/incidents/ HTTP/1.1" 200 OK
web_1        | ERROR:dispatch.messaging.email.utils:internal/modules/cjs/loader.js:983
web_1        |   throw err;
web_1        |   ^
web_1        | 
web_1        | Error: Cannot find module '../lib/index'
web_1        | Require stack:
web_1        | - /usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml
web_1        |     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
web_1        |     at Function.Module._load (internal/modules/cjs/loader.js:862:27)
web_1        |     at Module.require (internal/modules/cjs/loader.js:1042:19)
web_1        |     at require (internal/modules/cjs/helpers.js:77:18)
web_1        |     at Object.<anonymous> (/usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml:3:1)
web_1        |     at Module._compile (internal/modules/cjs/loader.js:1156:30)
web_1        |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
web_1        |     at Module.load (internal/modules/cjs/loader.js:1000:32)
web_1        |     at Function.Module._load (internal/modules/cjs/loader.js:899:14)
web_1        |     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12) {
web_1        |   code: 'MODULE_NOT_FOUND',
web_1        |   requireStack: [
web_1        |     '/usr/local/lib/python3.8/site-packages/dispatch/static/dispatch/node_modules/.bin/mjml'
web_1        |   ]
web_1        | }
web_1        | 
web_1        | ERROR:dispatch.decorators:MJML template processing failed.
web_1        | Traceback (most recent call last):
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/decorators.py", line 37, in wrapper
web_1        |     result = func(*args, **kwargs)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/report/flows.py", line 71, in create_tactical_report
web_1        |     send_tactical_report_to_tactical_group(incident_id, tactical_report, db_session)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/report/messaging.py", line 82, in send_tactical_report_to_tactical_group
web_1        |     plugin.instance.send(
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/decorators.py", line 58, in wrapper
web_1        |     result = func(*args, **kwargs)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/decorators.py", line 74, in wrapper
web_1        |     return func(*args, **kwargs)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/plugins/dispatch_google/gmail/plugin.py", line 94, in send
web_1        |     message_body = create_message_body(notification_template, notification_type, **kwargs)
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/messaging/email/utils.py", line 87, in create_message_body
web_1        |     return render_html(template.render(**kwargs))
web_1        |   File "/usr/local/lib/python3.8/site-packages/dispatch/messaging/email/utils.py", line 103, in render_html
web_1        |     raise Exception("MJML template processing failed.")
web_1        | Exception: MJML template processing failed.
web_1        | INFO:     172.20.0.1:64212 - "GET /api/v1/incidents/?q=&page=1&itemsPerPage=10&sortBy[]=reported_at&descending[]=true HTTP/1.1" 200 OK
```

--- a/.dockerignore
+++ b/.dockerignore
@@ -24,7 +24,7 @@ package.json
 /build
 /env
 /tmp
-/node_modules/
+node_modules
 /wheelhouse
 /test_cli/
 .idea/

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -8,73 +8,16 @@ LABEL org.opencontainers.image.source="https://github.com/netflix/dispatch"
 LABEL org.opencontainers.image.vendor="Netflix, Inc."
 LABEL org.opencontainers.image.authors="oss@netflix.com"
 
+# Get and set up Node for front-end asset building
 RUN apt-get update && apt-get install -y --no-install-recommends \
-    # Needed for GPG
-    dirmngr \
-    gnupg2 \
     # Needed for fetching stuff
     ca-certificates \
     wget \
     && rm -rf /var/lib/apt/lists/*
 
-# Fetch trusted keys
-RUN for key in \
-    # gosu
-    B42F6819007F00F88E364FD4036A9C25BF357DD4 \
-    # tini
-    595E85A6B1B4779EA4DAAEC70B588DFF0527A9B7 \
-    # Node - gpg keys listed at https://github.com/nodejs/node
-    94AE36675C464D64BAFA68DD7434390BDBE9B9C5 \
-    FD3A5288F042B6850C66B31F09FE44734EB7990E \
-    71DCFD284A79C3B38668286BC97EC7A07EDE3FC1 \
-    DD8F2338BAE7501E3DD5AC78C273792F7D83545D \
-    C4F0DFFF4E8C1A8236409D08E73BC641CC11F4C8 \
-    B9AE9905FFD7803F25714661B63B535A4C206CA9 \
-    77984A986EBC2AA786BC0F66B01FBB92821C587A \
-    8FCCA13FEF1D0C2E91008E09770F7A9A5AE15600 \
-    4ED778F539E3634C779C87C6D7062848A1AB005C \
-    A48C2BEE680E841632CD4E44F07496B3EB3C1762 \
-    B9E2F5981AA6E0CD28160D9FF13993A75599653C \
-    ; do \
-    # Let's try several servers to ensure we get all the keys
-    for server in \
-    keys.openpgp.org \
-    keyserver.ubuntu.com \
-    hkps.pool.sks-keyservers.net \
-    ; do \
-    gpg2 --batch --keyserver "$server" --recv-keys "$key"; \
-    done \
-    done
-
-# grab gosu for easy step-down from root
-ENV GOSU_VERSION 1.12
 RUN set -x \
-    && dpkgArch="$(dpkg --print-architecture | awk -F- '{ print $NF }')" \
-    && wget -O /usr/local/bin/gosu "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch" \
-    && wget -O /usr/local/bin/gosu.asc "https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc" \
-    && gpg2 --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu \
-    && rm -f /usr/local/bin/gosu.asc \
-    && chmod +x /usr/local/bin/gosu
-
-# grab tini for signal processing and zombie killing
-ENV TINI_VERSION 0.18.0
-RUN set -x \
-    && wget -O /usr/local/bin/tini "https://github.com/krallin/tini/releases/download/v$TINI_VERSION/tini" \
-    && wget -O /usr/local/bin/tini.asc "https://github.com/krallin/tini/releases/download/v$TINI_VERSION/tini.asc" \
-    && gpg --batch --verify /usr/local/bin/tini.asc /usr/local/bin/tini \
-    && rm -f /usr/local/bin/tini.asc \
-    && chmod +x /usr/local/bin/tini
-
-# Get and set up Node for front-end asset building
-COPY .nvmrc /usr/src/dispatch/
-RUN cd /usr/src/dispatch \
-    && export NODE_VERSION="$(cat .nvmrc)" \
-    && wget "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz" \
-    && wget "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \
-    && gpg --batch --verify SHASUMS256.txt.asc \
-    && grep " node-v$NODE_VERSION-linux-x64.tar.gz\$" SHASUMS256.txt.asc | sha256sum -c - \
-    && tar -xzf "node-v$NODE_VERSION-linux-x64.tar.gz" -C /usr/local --strip-components=1 \
-    && rm -f "node-v$NODE_VERSION-linux-x64.tar.gz" SHASUMS256.txt.asc
+    && wget --quiet -O - https://deb.nodesource.com/setup_12.x | bash - \
+    && apt-get install -y nodejs
 
 ARG SOURCE_COMMIT
 ENV DISPATCH_BUILD=${SOURCE_COMMIT:-unknown}
@@ -109,8 +52,6 @@ LABEL org.opencontainers.image.authors="oss@netflix.com"
 # add our user and group first to make sure their IDs get assigned consistently
 RUN groupadd -r dispatch && useradd -r -m -g dispatch dispatch
 
-COPY --from=sdist /usr/local/bin/gosu /usr/local/bin/tini /usr/local/bin/node /usr/local/bin/
-
 # Sane defaults for pip
 ENV PIP_NO_CACHE_DIR=off \
     PIP_DISABLE_PIP_VERSION_CHECK=1 \
@@ -120,6 +61,9 @@ ENV PIP_NO_CACHE_DIR=off \
 RUN set -x \
     && echo "deb http://apt.postgresql.org/pub/repos/apt buster-pgdg main" > /etc/apt/sources.list.d/pgdg.list \
     && wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
+
+RUN set -x \
+    && wget --quiet -O - https://deb.nodesource.com/setup_12.x | bash -
 
 COPY --from=sdist /dist/*.whl /tmp/dist/
 RUN set -x \
@@ -134,9 +78,10 @@ RUN set -x \
     # These are also installed last as they are needed
     # during container run and can have the same deps w/
     && apt-get install -y --no-install-recommends \
-    pkg-config postgresql-client-12 \
+    pkg-config postgresql-client-12 nodejs \
     && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
+    && rm -rf /var/lib/apt/lists/* \
+    && npm install mjml
 
 EXPOSE 8000
 VOLUME /var/lib/dispatch/files

